src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx

```diff
@@ -118,23 +135,52 @@ export const WorkspacePermissionSettingInput = ({
   );
 
   const handlePermissionModeOptionChange = useCallback(
-    (id: string) => {
-      if (optionIdToWorkspacePermissionModesMap[id]) {
-        onPermissionModesChange([...optionIdToWorkspacePermissionModesMap[id]], index);
+    (option: Array<EuiComboBoxOptionOption<any>>) => {
+      if (option.length > 0) {
+        const id = option[0].value;
+        if (optionIdToWorkspacePermissionModesMap[id]) {
+          onPermissionModesChange([...optionIdToWorkspacePermissionModesMap[id]], index);
+        }
       }
     },
     [index, onPermissionModesChange]
   );
 
+  const toggleTypeList = () => {
+    if (!isTypeListOpen) {
+      if (type === WorkspacePermissionItemType.User) {
+        setTypeOptions([{ ...typeOptions[0], checked: 'on' }, typeOptions[1]]);
+      } else {
+        setTypeOptions([typeOptions[0], { ...typeOptions[1], checked: 'on' }]);
+      }
+    }
```

Comment:
Why update the options when the popover is closed?

Reply:
because the current type should be selected in the popover list

Reply:
Oh, you're right, I missed that part, then you need to mark it with `checked`